### DISCUSSION TITLE
Changing the working directory for an extension

### DISCUSSION THREAD

**Discussion by openscript** [microsoft/vscode-discussions#466] [Extension Development QnA] [answered]
I'm developing an extension, where I dynamically import from the workspace. To allow relative paths inside the workspace being resolved, I set the working directory, so I can `import('./some/relative/path.js')`:

```ts
  process.chdir(workspacePath);
```

My questions are:
1. As the extension run inside their own process, other extensions and VS Code processes shouldn't be affected from this, right?
1. Are there environments (local extension host, web extension host, remote extension host) which don't allow `process.chdir()`?

**Comment by r3m0t** [comment, marked as answer]
All extensions of the same type (eg local extensions) run in the same extension host so if you import the user's code from their workspace, you might crash your extension and all the other extensions as well.

**Reply by openscript** [reply]
Are you sure about this? Here it looks like each extension has its individual process.

![image](https://user-images.githubusercontent.com/1105080/219467323-253dddee-e9a1-4ef0-a129-984755845892.png)

**Reply by r3m0t** [reply]
The fact you can get extension APIs from other extensions shows they are all running in the same process. It looks like you have one local extension host with the Docker extension installed, and it is running many Docker subprocesses, maybe for different features, but I think only one of the subprocesses is an extension host.
